sdk/test/trace/batch_span_processor_test.cc

```diff
@@ -291,4 +304,71 @@ TEST_F(BatchSpanProcessorTestPeer, TestScheduleDelayMillis)
   }
 }
 
+#if defined(_UNISTD_H) && defined(_SYS_WAIT_H)
+#  ifndef __SANITIZE_THREAD__
+TEST_F(BatchSpanProcessorTestPeer, TestForkHandlers)
+{
+  std::shared_ptr<std::atomic<bool>> is_shutdown(new std::atomic<bool>(false));
+  std::shared_ptr<std::atomic<bool>> is_export_completed(new std::atomic<bool>(false));
+  std::shared_ptr<std::vector<std::unique_ptr<SpanData>>> spans_received(
+      new std::vector<std::unique_ptr<SpanData>>);
+
+  const std::chrono::milliseconds export_delay(0);
+  const std::chrono::milliseconds schedule_delay_millis(1000);
+
+  auto batch_processor = GetMockProcessor(spans_received, is_shutdown, is_export_completed,
+                                          export_delay, schedule_delay_millis);
+
+  auto provider = nostd::shared_ptr<opentelemetry::trace::TracerProvider>(
+      new sdk::trace::TracerProvider(batch_processor));
+
+  auto tracer = provider->GetTracer("Test tracer");
+
+  // Create a span in the parent process
+  auto span = tracer->StartSpan("Span");
+  span->End();
+
+  std::this_thread::sleep_for(schedule_delay_millis + std::chrono::milliseconds(50));
+  // The span in the parent process will now be exported
+  EXPECT_TRUE(is_export_completed->load());
+
+  pid_t child_pid = fork();
+
+  if (child_pid == 0)
+  {
+    // CHILD PROCESS
+
+    // Reset MockExporter private members.
+    // This will automatically be handled after a ForkAwareExporter is implemented.
+    spans_received->clear();
+    *is_export_completed = false;
+
+    // Generate some spans
+    for (int i = 0; i < 3; ++i)
+    {
+      auto span = tracer->StartSpan("Span");
+    }
+
+    std::this_thread::sleep_for(schedule_delay_millis + std::chrono::milliseconds(50));
+
+    // Spans should now be exported in the child process
+    EXPECT_TRUE(is_export_completed->load());
+    EXPECT_EQ(3, spans_received->size());
+
+    batch_processor->Shutdown();
+    // End child process
+    std::exit(0);
+  }
+
+  // We only created one span in the parent process
+  EXPECT_EQ(1, spans_received->size());
```

Comment:
All the `EXPECT` statements pass (when I run it locally on my machine), but I believe the CircleCI checks are not  checking the `EXPECT` statements in the child process.

Adding a shared memory segment (using `mmap`) to make sure that CircleCI checks whether or not the tests in the child process pass.